databricks_cli/sdk/service.py

```diff
@@ -821,21 +821,24 @@ def deploy(self, pipeline_id=None, id=None, name=None, storage=None, configurati
                 raise TypeError('Expected databricks.Filters() or dict for field filters')
         return self.client.perform_query('PUT', '/pipelines/{pipeline_id}', data=_data, headers=headers)
 
+    def deploy_spec(self, pipeline_id=None, spec={}, headers=None):
+        return self.client.perform_query('PUT', '/pipelines/{}'.format(pipeline_id), data=spec, headers=headers)
+
     def delete(self, pipeline_id=None, headers=None):
         _data = {}
         if pipeline_id is not None:
             _data['pipeline_id'] = pipeline_id
-        return self.client.perform_query('DELETE', '/pipelines/{pipeline_id}', data=_data, headers=headers)
+        return self.client.perform_query('DELETE', '/pipelines/{}'.format(pipeline_id), data=_data, headers=headers)
```

Comment:
this is an unrelated question because existing code did this: Why are we passing the pipeline_id in the URL as well as in the body of the request? Is it required in the body?

Reply:
Not required, but that's just the generated way. Seems to be fine to send it in the body if it's the same.